### ISSUE TITLE
TLA+ Toolkit 1.5.3 on JDK 9 on macOS does not respond to mouse clicks or keystrokes.

### ISSUE THREAD

**Issue by nelsonjchen** [tlaplus/tlaplus#74]
In a fresh MacOS 10.12.6 VM: https://www.youtube.com/watch?v=z8O6q3du86I&feature=youtu.be

Version: 1.5.3 

The only keystroke it responded to, and arguably wasn't it to handle, was Cmd+Q in the app selector to quit the app. I can reproduce this on the host too but I like to record issues in reproducible environments.

I guess users should be advised to use JDK 8? I did download Eclipse Oxygen and it still seemed to work fine with JDK 9 and I did notice an Eclipse version bump in the repo so maybe it's resolved on the edge. Unfortunately, I can't tell because I wasn't able to build the IDE on my Mac to give the latest bleeding edge a try.

**Comment by lemmy** [comment]
On 24.09.2017 07:32, Nelson Chen wrote:
> In a fresh MacOS 10.12.6 VM:
> https://www.youtube.com/watch?v=z8O6q3du86I&feature=youtu.be
> 
> The only keystroke it responded to, and arguably wasn't it to handle,
> was Cmd+Q in the app selector to quit the app. I can reproduce this on
> the host too but I like to record issues in reproducible environments.
> 
> I guess users should be advised to use JDK 8? I did download Eclipse
> Oxygen and it still seemed to work fine with JDK 9 and I did notice an
> Eclipse version bump in the repo so maybe it's resolved on the edge.
> Unfortunately, I can't tell because I wasn't able to build the IDE on my
> Mac to give the latest bleeding edge a try.

Hi Nelson,

thanks for your bug report. We have indeed made several changes -
including an upgrade to Eclipse Oxygen - in preparation of Java 9.
Generally, you don't have to build the Toolbox yourself. You can find
the edge (nightly) builds at [1]. However, the current nightly build
crashes on Mac with JDK 9. We plan to fix this soon. In the meantime
please stick to JDK 8 until we have sorted things out.

By the way, what errors did you run into in your attempt to build the
Toolbox? Building it is supposed to work out of the box with "mvn clean
verify" (optinally skip the tests with "mvn clean verify
-Dmaven.test.skip=true") at the root of the git repository [2]. If you
want a full-fledged Eclipse IDE setup for Toolbox development, the
necessary steps are listed at [3].

Thanks
Markus

[1] http://tla.msr-inria.inria.fr/tlatoolbox/ci/products/
[2] https://github.com/tlaplus/tlaplus.git
[3] https://github.com/tlaplus/tlaplus/tree/master/general/ide

**Comment by nelsonjchen** [comment]
Thanks for looking at this. I had a feeling the recent commits were related. On that note, I actually tried Java 9 after some version of Java 8 also failed to work. In fact, i think I was running a pre-release version of Java 8, b152 or something. I only did that because I didn't want to see some annoying cosmetic error in one of my IntelliJ-based building process of some other software. Pre-release or non-production JDKs **bite** and this is one of them. 🤦‍♂️ .  

Ick! I missed that second paragraph for the Toolbox development part that your third link links to. I might put in a PR to make correct that link in the README.md at the root into a link to the rendered markdown instead of the raw version.  Also, whatever command I was running didn't skip the tests. I would have figured that out if I did not overlook that IDE readme. Thanks for pointing that out.
